### ISSUE TITLE
Temp files triggers error in watch tasks

### ISSUE THREAD

**Issue by AndyOGo** [idok/scss-lint-plugin#45]
### Versions

scss-lint version: I can't tell, cause it seems to bundled with PHPStorm
You can get this information from executing `scss-lint --version` at the command line.

OS: Mac OS X El Capitan 10.11.6
IDEA product and version: PHPStorm 2016.2

### Description

SCSS-lint tries to lint JS files.

> Error: ENOENT: no such file or directory, open '/Users/andy/dev/style-guide/jquery/register-plugin_scsslint_tmp802830433474418013.js'
>     at Error (native)

There seems to be an related issue, but it does not seem to be resolved:
https://github.com/idok/scss-lint-plugin/issues/13

### Steps to Reproduce

1. Setup Gulp watch for JavaScript bundling, minification, etc.
2. Edit JS files in PHPStorm
3. SCSS-Lint generates temporary files for JS and immediatly removes them
4. This triggers gulp watch
5. Gulp watch dies because he can't find the file again.

**Expected behavior:**

- SCSS-lint should never lint javascript files
- SCSS-lint should never create temporary files inside source folders

**Actual behavior:**
- SCSS-lint lints javascript files
- SCSS-lint create temporary files inside source folders

**Comment by AndyOGo** [comment]
Any updates on this?

**Comment by AndyOGo** [comment]
@idok 
Did you have any chance to look into this.
It seems to be really weird for two reasons:
- it lints `.js` files with scsslint...
- tmp files are stored inside source folders, imho they should be in a seperate `tmp/` dir

`tasks/docs-scripts_scsslint_tmp1537225672253761620.js`
`tasks/scripts-bundle_scsslint_tmp4377624646903046346.js`

I tried with other IDE's and only InteliJ based IDE suffer from this illness

**Comment by AndyOGo** [comment]
I quickly search you repo for `scsslint` and the following two hints:
https://github.com/idok/scss-lint-plugin/blob/63215554eb3c194478a7617ca611a1df7281d92f/intellij-common/src/com/wix/ThreadLocalActualFile.java#L25
https://github.com/idok/scss-lint-plugin/blob/63215554eb3c194478a7617ca611a1df7281d92f/scss-lint-plugin/src/com/scss/ThreadLocalActualFile.java#L24

**Comment by jillesme** [comment]
I have this issue, too

**Comment by laurenbrandstein** [comment]
Me too. Or me three. Or four. This breaks the watch and requires rerunning gulp. Inconvenient.

**Comment by aspirinchaos** [comment]
@AndyOGo 
have you found how to disable this plugin?

**Comment by AndyOGo** [comment]
@aspirinchaos 
Unfortunately no, I still don't know how to disable this 😞 
Though I can confirm that in case of Gulp this issue can be workarounded by adding GLOB exclusion `'!**/*scsslint*js'` as asuggested already by @laurenbrandstein 

Anyway I switched to webpack and didn't experience this bug there...

**Comment by AndyOGo** [comment]
I have created an Issue now directly at JetBrains
https://youtrack.jetbrains.com/issue/WI-37757

**Comment by idok** [comment]
Hi Andy,
Do you have any other plugins installed?

**Comment by AndyOGo** [comment]
Hi @idok 
No, nothing related to SCSS.

Btw. I already found the bug [here](https://github.com/idok/scss-lint-plugin/issues/45#issuecomment-247543028):

> I quickly searched your repo for `scsslint` and the following two hints came up:
>
> ``` java
> private static final String SCSS_LINT_TMP = "_scsslint_tmp";
> ```
> 
> https://github.com/idok/scss-lint-plugin/blob/63215554eb3c194478a7617ca611a1df7281d92f/intellij-common/src/com/wix/ThreadLocalActualFile.java#L25
> 
> And the file creation here:
> 
> ``` java
> // try to create a temp file next to original file
> retFile = File.createTempFile(this.baseName + SCSS_LINT_TMP, this.extension, new File(originalFile.getParent().getPath()));
> ```
> 
> https://github.com/idok/scss-lint-plugin/blob/63215554eb3c194478a7617ca611a1df7281d92f/intellij-common/src/com/wix/ThreadLocalActualFile.java#L74-L75

**Comment by idok** [comment]
SCSS Lint plugin should and as far as I know only lints scss files.
Do you have eslint plugin installed?
If you uninstall SCSS Lint plugin do you still see temp js files?

**Comment by AndyOGo** [comment]
@idok 
Sure I use ESLint, but installed as node package.
Unfortunately I reported this issue about a year ago, meanwhile I changed my stack a lot.
to wepback, stylelint, ... so I can't supply you with any actual logs... sorry

**Comment by softmantk** [comment]
This issue still exist! My dev runner nodemon is restarting and throwing errors.
